flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputGateMetrics.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.consumer;
+
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.metrics.MetricGroup;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+public class InputGateMetrics {
+
+	private final SingleInputGate inputGate;
+
+	private long lastTotal = -1;
+
+	private int lastMin = -1;
+
+	private int lastMax = -1;
+
+	private float lastAvg = -1.0f;
+
+	// ------------------------------------------------------------------------
+
+	private InputGateMetrics(SingleInputGate inputGate) {
+		this.inputGate = checkNotNull(inputGate);
+	}
+
+	// ------------------------------------------------------------------------
+
+	// these methods are package private to make access from the nested classes faster 
+
+	long refreshAndGetTotal() {
+		long total;
+		if ((total = lastTotal) == -1) {
+			refresh();
```

Comment:
I guess the main reason for this was performance in case all (up to) 4 metrics are requested. What's the preferred way of exposing these kind of metrics? Should I gather all 4 in a custom object and enclose that one in a Gauge? How could the metrics then be displayed, e.g. in the web interface?